src/utils/nvim.ts

```diff
@@ -0,0 +1,38 @@
+import { NeovimClient } from "neovim";
+
+import { ILogger, LogLevel } from "../logger";
+
+/**
+ * Wrap nvim callAtomic and check for any errors in result
+ * @param client
+ * @param requests
+ * @param logger
+ * @param prefix
+ */
+export async function callAtomic(
```

Comment:
(off topic) we could eliminate this and `executeLua` instead

Reply:
Interesting - I haven't been deep into neovim internals before, so this is a bit new to me. Is there any chance of only some operations completing if were to convert a block like this to use many statements in `executeLua`? I only ask because the docs for the neovim client say that `callAtomic` has some guarantees around atomicity, and I don't see that for `executeLua`

https://github.com/vscode-neovim/vscode-neovim/blob/master/src/buffer_manager.ts#L605-L620

Reply:
The "atomicity" of https://neovim.io/doc/user/api.html#nvim_call_atomic() is mostly just about performing multiple things in a single RPC request. Since `nvim_exec_lua` was introduced it is no longer needed.